src/async_operator/facet.rs

```diff
@@ -0,0 +1,41 @@
+use crate::{operator::facet::Facet, AsyncOperator};
+use core::task::{Context, Poll};
+use futures::{
+    future::{join, BoxFuture},
```

Comment:
It seems that we need to enable `futures/alloc` feature to use `BoxFuture` in `no_std`